crates/python/src/qvm/mod.rs

```diff
@@ -52,15 +67,79 @@ impl PyQvmClient {
         })
     }
 
+    #[cfg(feature = "libquil")]
+    #[staticmethod]
+    fn new_libquil() -> PyResult<Self> {
+        Ok(Self {
+            inner: QvmClient::Libquil(qvm::libquil::Client {}),
+        })
+    }
+
+    #[cfg(not(feature = "libquil"))]
+    #[staticmethod]
+    fn new_libquil() -> PyResult<Self> {
+        Err(PyRuntimeError::new_err(
+            "Cannot create a libquil QVM client as feature is not enabled.",
+        ))
+    }
+
     #[getter]
     fn qvm_url(&self) -> PyResult<String> {
-        let QvmClient::Http(client) = &self.inner;
-        Ok(client.qvm_url.to_string())
+        match &self.inner {
+            QvmClient::Http(client) => Ok(client.qvm_url.to_string()),
+            #[cfg(feature = "libquil")]
+            QvmClient::Libquil(_) => Ok("".into()),
+        }
     }
 }
 
-wrap_error!(RustQvmError(qcs::qvm::Error));
-py_wrap_error!(api, RustQvmError, QVMError, PyRuntimeError);
+#[async_trait::async_trait]
+impl qvm::Client for QvmClient {
```

Comment:
It looks like you're now using `.as_client()` wherever there used to be a direct conversion to `qvm::http::HttpClient`, so why is this new implementation necessary? Or, conversely, why not get rid of some of the `as_client()` calls in other modules, and rely on this implementation?

Reply:
FWIW, that had been my suggestion in the beginning: use `as_client()` in this implementation and then rely on this implementation everywhere else.